packages/office-addin-telemetry/src/telemetryJsonData.ts

```diff
@@ -0,0 +1,111 @@
+
+import * as chalk from "chalk";
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+import { telemetryLevel } from "./officeAddinTelemetry";
+const telemetryJsonFilePath: string = path.join(os.homedir(), "/officeAddinTelemetry.json");
+
+/**
+ * Allows developer to check to check if the program has already prompted before
+ * @param groupName Group name of the telemetry object
+ * @param jsonFilePath Optional path to the json config file
+ * @returns Boolean of whether the program should prompt
+ */
+export function promptForTelemetry(groupName: string, jsonFilePath = telemetryJsonFilePath): boolean {
+    try {
+        const jsonData: any = readTelemetryJsonData(jsonFilePath);
+        if (jsonData) {
+            return !groupNameExists(jsonData, groupName);
+        }
+        return true;
+    } catch (err) {
+        console.log(chalk.default.red(err));
+    }
+}
+/**
+ * Allows developer to add or modify a specific property to the group
+ * @param groupName Group name of property
+ * @param property Property that will be created or modified
+ * @param value Property's value that will be assigned
+ * @param jsonFilePath Optional path to the json config file
+ */
+export function modifyTelemetryJsonData(groupName: string, property: any, value: any, jsonFilePath = telemetryJsonFilePath): void {
+    try {
+        if (fs.existsSync(jsonFilePath)) {
+            const telemetryJsonData = readTelemetryJsonData(jsonFilePath);
+            if (groupNameExists(telemetryJsonData, groupName)) {
+                telemetryJsonData.telemetryInstances[groupName][property] = value;
+                fs.writeFileSync(jsonFilePath, JSON.stringify((telemetryJsonData), null, 2));
+            }
+        }
+    } catch {
+        console.log(chalk.default.red("Could not modify property!\n Make sure the group name and file exists!"));
+    }
+}
+/**
+ * Reads data from the telemetry json config file
+ * @param jsonFilePath Optional path to the json config file
+ * @returns Parsed object from json file if it exists
+ */
+export function readTelemetryJsonData(jsonFilePath = telemetryJsonFilePath): any {
+    if (fs.existsSync(jsonFilePath)) {
+        const jsonData = fs.readFileSync(jsonFilePath, "utf8");
+        return JSON.parse(jsonData.toString());
+    }
+}
+/**
+ * Returns telemetry level of telemetry object
+ * @param groupName Group name to search for in the specified json data
+ * @param jsonFilePath Optional path to the json config file
+ * @returns Telemetry level specific to the group name
+ */
+export function readTelemetryLevel(groupName: string, jsonFilePath = telemetryJsonFilePath): string {
```

Comment:
the return type here should be telemetryLevel and not string.  That allows for us to do the following:

telemetryObject.telemetryLevel = telemetryJsonData.readTelemetryLevel(telemetryObject.groupName);